lte/gateway/c/session_manager/test/test_session_manager_handler.cpp

```diff
@@ -101,22 +101,26 @@ class SessionManagerHandlerTest : public ::testing::Test {
     rule_store->insert_rule(rule);
   }
 
-  // This function should always be called at the beginning of the test to
-  // prevent unexpected SessionD <-> PipelineD syncing logic mid-test.
-  void send_empty_table_and_wait_for_setup(SetupFlowsResult_Result res) {
+  void send_empty_table() {
     RuleRecordTable empty_table;
     // epoch indicates the last PipelineD service start time
     empty_table.set_epoch(DEFAULT_PIPELINED_EPOCH);
     grpc::ServerContext context;
     session_manager->ReportRuleStats(
         &context, &empty_table,
         [this](grpc::Status status, Void response_out) {});
+  }
```

Comment:
wouldnt make more sense to add the 

```
    evb->loopOnce();
    evb->loopOnce();
```
to this function `send_empty_table`?
I think the action that requires some event loops to happen is the `ReportRuleStats`

Reply:
The looping has to be done after the setup expectation. it is a bit ugly :/